autograder-extra/pom.xml

```diff
@@ -21,6 +21,10 @@
 
     <dependencies>
         <!-- Internal -->
+        <dependency>
+            <groupId>de.firemage.autograder</groupId>
+            <artifactId>autograder-api</artifactId>
+        </dependency>
```

Comment:
Same here .. this would ensure that the API is not a transitive dependency.. You could also think about adding the extra jar to the provided artifacts and make a flag in the API to load it as well.